README.md

```diff
@@ -1,125 +1,206 @@
+### Valid
+An `Order` is only valid if 
+* non-cancelled, 
+* non-expired, 
+* not-fully-filled and 
+* passes signature validation.
+### Invalid Order
+An order that has been cancelled / filled / expired or insolvent.
+#### Cancelled
+Only an `Order` may be cancelled. If it is, an entry into the on-chain _cancelled_ mapping is stored. No `Agreements` will then be accepted for this order. 
+#### Full
+The order has been filled in entirety but one or more fills.
+#### Temporarily Insolvent
+When an order is placed that uses a Vault's entire balance, Swivel will conduct a secondary check for solvency to ensure that no insolvent orders enter the orderbook. During this intermediary period, the order will be marked as _temporarily insolvent_.
+#### Insolvent
+If an order is placed and the underlying assets are subsequently not available, the order is marked as insolvent.
+#### Expired
+An Order which has aged past its Duration. Like a cancelled Order, it may no longer be filled.
+### Event
+An observable occurance that is recorded either on-chain or off-chain in Swivel's event store.
+### Message
+A visible message that is sent from an event store that can be listened for by a user or service.
+### Signature
+As ECDSA cryptographic signature following the EIP-712 standard which we use to verify the authenticity of an `Order`. You will see the `Components` of a Signature (V, R, and S) used throughout the protocol. The Hash and Sig solidity libraries exist to service this functionality. It should be noted and enforced that `Components` are derived from a `Signature` and that the two are not equivalent. a `Signature` is a 64-bytes and its derived `Components` are a _tuple_ of length 3 consisting of two 32-bytes, the canonical _R_ and _S_,  and a single uint _V_.
 
-### Transfer
-Terms such as transfer, transfers  and transfer from when referencing specific functionality of the Underlying token.
-As with approve, specificy which token if necessary, _uToken transfer_ etc...
+An Order's signature is considered `valid` if the extracted public key matches that of the maker.
+### EIP-712 Domain
+Mentioned here as the term `Domain` has very specific Domain Driven Design implications. You may see or hear the term domain intermingled with other verbage for the EIP-712 standard such as:
+* TypeHash
+* Separator
+* Message
 
-### Balance
-Amount of the underlying token a given public key owns. Specify which token to avoid confusion.
+Any use of these terms, specifically _domain_ should be prefixed with EIP-712 to prevent language degradation and the confusion that follows it.
 
-## Compound Token
-A CErc20 at an address passed to the constructor of the Swivel contract at deployement.
-* Perhaps pass an array of underlying token addresses + array of CERC20 addresses to create a public mapping of address -> address 
+Implementation of Signature and Components are each tightly coupled to the EIP-712 standard.
+### Initiate Event
```

Comment:
the initiate event has many arguments in V2, likely more than you want to list:

`event Initiate(bytes32 indexed key, bytes32 hash, address indexed maker, bool vault, bool exit, address indexed sender, uint256 amount, uint256 filled);`

so, the thing to do might be to simply link to the contract for event argument specifics.